plugins/azure/fix_plugin_azure/resource/keyvault.py

```diff
@@ -0,0 +1,415 @@
+    soft_delete_retention_in_days: Optional[int] = field(default=None, metadata={'description': 'Soft deleted data retention days. When you delete an HSM or a key, it will remain recoverable for the configured retention period or for a default period of 90 days. It accepts values between 7 and 90.'})  # fmt: skip
+    status_message: Optional[str] = field(default=None, metadata={"description": "Resource Status Message."})
+    tenant_id: Optional[str] = field(default=None, metadata={'description': 'The Azure Active Directory tenant ID that should be used for authenticating requests to the managed HSM pool.'})  # fmt: skip
+
+
+@define(eq=False, slots=False)
+class AzureKey(MicrosoftResource):
+    kind: ClassVar[str] = "azure_key"
+    # collected via AzureKeyVault
+    mapping: ClassVar[Dict[str, Bender]] = {
+        "id": S("id"),
+        "tags": S("tags", default={}),
+        "name": S("name"),
+        "attributes": S("properties", "attributes") >> Bend(AzureKeyAttributes.mapping),
+        "curve_name": S("properties", "curveName"),
+        "key_ops": S("properties", "keyOps"),
+        "key_size": S("properties", "keySize"),
+        "key_uri": S("properties", "keyUri"),
+        "key_uri_with_version": S("properties", "keyUriWithVersion"),
+        "kty": S("properties", "kty"),
+        "release_policy": S("properties", "release_policy") >> Bend(AzureKeyReleasePolicy.mapping),
+        "rotation_policy": S("properties", "rotationPolicy") >> Bend(AzureKeyVaultRotationPolicy.mapping),
+    }
+    attributes: Optional[AzureKeyAttributes] = field(default=None, metadata={'description': 'The object attributes managed by the Azure Key Vault service.'})  # fmt: skip
+    curve_name: Optional[str] = field(default=None, metadata={'description': 'The elliptic curve name. For valid values, see JsonWebKeyCurveName.'})  # fmt: skip
+    key_ops: Optional[List[str]] = field(default=None, metadata={"description": ""})
+    key_size: Optional[int] = field(default=None, metadata={'description': 'The key size in bits. For example: 2048, 3072, or 4096 for RSA.'})  # fmt: skip
+    key_uri: Optional[str] = field(default=None, metadata={'description': 'The URI to retrieve the current version of the key.'})  # fmt: skip
+    key_uri_with_version: Optional[str] = field(default=None, metadata={'description': 'The URI to retrieve the specific version of the key.'})  # fmt: skip
+    kty: Optional[str] = field(default=None, metadata={'description': 'The type of the key. For valid values, see JsonWebKeyType.'})  # fmt: skip
+    release_policy: Optional[AzureKeyReleasePolicy] = field(default=None, metadata={"description": ""})
+    rotation_policy: Optional[AzureKeyVaultRotationPolicy] = field(default=None, metadata={"description": ""})
+
+
+@define(eq=False, slots=False)
+class AzureKeyVault(MicrosoftResource):
```

Comment:
Can we add `BaseAccessKey` inheritance for `AzureKeyVault`?

Reply:
No the vault is used to keep secrets - not access keys.